restructuredtext_lint/ignore.py

```diff
@@ -0,0 +1,79 @@
+import contextlib
+
+from docutils.parsers import rst
+from docutils.parsers.rst import directives as rst_directives
+from docutils.parsers.rst import roles as rst_roles
+
+
+class _IgnoredDirective(rst.Directive):
+    """Stub for unknown directives.
+
+    See: http://repo.or.cz/w/docutils.git/blob/1976ba91eff979abc3e13e5d8cb68324833af6a0:/docutils/parsers/rst/__init__.py#l194  # noqa
+
+    See: https://github.com/myint/rstcheck/blob/v1.1.1/rstcheck.py#L217
+    """
+    has_content = True
+
+    def run(self):
+        return []
+
+
+def _ignore_role(name, rawtext, text, lineno, inliner,
+                 options=None, content=None):
+    """Stub for unknown roles.
+
+    See: http://repo.or.cz/w/docutils.git/blob/1976ba91eff979abc3e13e5d8cb68324833af6a0:/docutils/parsers/rst/roles.py
+
+    See: https://github.com/myint/rstcheck/blob/v1.1.1/rstcheck.py#L228
+    """
+    return ([], [])
+
+
+def register_ignores(directives, roles):
+    """Register ignoreable sphinx directives & roles.
+
+    :param list directives: directives to ignore
+    :param list roles: roles to ignore
+
+    :rtype (list, list): tuple of directives and roles.
+    """
+    for directive in directives:
+        rst_directives.register_directive(directive, _IgnoredDirective)
+    for role in roles:
+        rst_roles.register_local_role(role, _ignore_role)
+    return (directives, roles)
```

Comment:
We shouldn't be returning anything from this function since the user already knows what they passed in. By leaving it as nothing for now, we leave room for future requests like returning all registered directives.